File: NRepository/NRepository.Webforms.VB/Controls/SampleControl.ascx.vb
Imports System.ComponentModel.Composition
Imports NRepository.UniversityBL.BL

Public Class SampleControl
    Inherits System.Web.UI.UserControl

    <Import>
    Public Property TestProvider As CourseProvider

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        If (TestProvider Is Nothing) Then
            Throw New ArgumentNullException("TestProvider")
        End If
    End Sub

End Class

File: NRepository/NRepository.WebForms/Controls/SampleControl.ascx.cs
using NRepository.UniversityBL.BL;
using System;
using System.ComponentModel.Composition;

namespace NRepository.WebForms.Controls
{
    public partial class SampleControl : System.Web.UI.UserControl
    {
        [Import]
        public CourseProvider TestProvider { get; set; }

        protected void Page_Load(object sender, EventArgs e)
        {
            if(TestProvider == null)
            {
                throw new ArgumentNullException("TestProvider");
            }
        }
    }
}